### PR TITLE
Break mobile app into multiple components with new features

### PR DESCRIPTION
**Pull request by chc5**

### CODE REVIEW

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import { NodeCameraView } from 'react-native-nodemediaclient';
+import { v4 as uuidv4 } from 'uuid';
+import queryString from 'query-string';
+
+/**
+ * A wrapper around react-native-nodemediaclient component
+ */
+class CameraPublisher extends React.Component<CameraProps, CameraState> {
+  // Initalized by nodemediaclient to get controller methods from library
+  vb: NodeMediaClientRef | undefined;
```
Comment (melvinheks) [inline src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:13]: visual basic

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { PermissionsAndroid, Alert } from 'react-native';
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (chc5) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:3]: There's no typescript module for react-native-nodemediaclient, so I ignored the typescript error that requires imports to have types

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+  /**
+   * Alerts the user about the status of the streaming app
+   * @param statusCode 2000 - Loading
+   *                   2001 - Started Streaming
+   *                   2002 - Error
+   *                   2004 - Closing
+   */
+  onStatus(statusCode: NodeMediaClientStatusCode) {
+    switch (statusCode) {
```
[the remark this answers is not in the document]
Comment (chc5) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:103]: what?

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx (hunk 1 above)
Comment (chc5) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:13]: yes

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chc5) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:3]: I'm gonna create one in an upcoming change

File: src/camera/mobile/components/camera-publisher/CameraPublisher.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (melvinheks) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:103]: image

[the remark this answers is not in the document]
Comment (chc5) [reply src/camera/mobile/components/camera-publisher/CameraPublisher.tsx:103]: yes

**Pull request merged**